tools/ci_build/build.py

```diff
@@ -2271,6 +2271,9 @@ def main():
 
     args = parse_arguments()
 
+    if os.getenv("ORT_BUILD_WITH_CACHE") == "1":
```

Comment:
There's an option because packaging pipeline shouldn't use caches.
The env is used when the build.py is encapsulated and it's not easy to pass the --use_cache.
Besides the env, we have to mount one dir to host cache dir.  
Is there better suggstions?

Reply:
@snnn, any more comments